pkg/kubelet/pod_workers.go

```diff
@@ -981,25 +981,29 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	// enforce the restriction that a grace period can only decrease and track whatever our value is,
 	// then ensure a calculated value is passed down to lower levels
 	gracePeriod := status.gracePeriod
-	// this value is bedrock truth - the apiserver owns telling us this value calculated by apiserver
-	if override := pod.DeletionGracePeriodSeconds; override != nil {
-		if gracePeriod == 0 || *override < gracePeriod {
+
+	// make a best effort to default this value to the pod's desired intent, in the event
```

Comment:
With the reorder of  checks, I believe we can improve the understandability of the blocks if we improve the comments.

Below is just a suggestion, but let me know what you think
```suggestion
	// defaults the grace period value to the pod's specified termination grace period seconds if present
	// this happens when kubelet provides no requested value or during graceful termination.
```

Reply:
Confimed the changes in the recent push. However the line below seems to be unchaned and does not connect correctly to the fixed line:

> 	// the kubelet provided no requested value (graceful termination?)

Just fixing my suggestion in the above comment, We can change that line to say something like:

>     // this happens when kubelet provides no requests to change this value or during graceful termination.

to make the line more coherent.